src/frontends/tensorflow/src/op/max_pool.cpp

```diff
@@ -14,56 +15,103 @@ namespace frontend {
+    }
+
+    // prepare input to MaxPool
+    convert_nhwc_to_nchw(is_nhwc, input);
 
-    bool is_nhwc = (tf_data_format == "NHWC") || (tf_data_format == "NDHWC");
+    auto max_pool_node = std::make_shared<ov::opset8::MaxPool>(input,
+                                                               strides,
+                                                               dilations,
+                                                               ov::Shape(pads_begin.begin(), pads_begin.end()),
+                                                               ov::Shape(pads_end.begin(), pads_end.end()),
+                                                               kernel_sizes,
+                                                               ov::op::RoundingType::FLOOR,
+                                                               auto_pad);
+    auto max_pool = max_pool_node->output(0);
+    ov::frontend::tensorflow::convert_nchw_to_nhwc(is_nhwc, max_pool);
+    ov::frontend::tensorflow::set_node_name(node.get_name(), max_pool.get_node_shared_ptr());
+    return {max_pool};
+}
+
+OutputVector translate_max_pool(const NodeContext& node, size_t spatial_dims_num) {
+    // MaxPool2D and MaxPool3D have ksize and strides as attributes
+    // retrieve attributes
+    auto strides = node.get_attribute<std::vector<int64_t>>("strides");
+    auto kernel_sizes = node.get_attribute<std::vector<int64_t>>("ksize");
+    return translate_max_pool_util(node, spatial_dims_num, kernel_sizes, strides);
+}
+
+OutputVector translate_max_pool_v2(const NodeContext& node) {
+    // MaxPoolV2 has ksize and strides as input parameters
+    TENSORFLOW_OP_VALIDATION(node, node.get_input_size() > 2, "MaxPoolV2 operation must have at least three inputs.");
+    auto ksize = node.get_input(1);
+    auto strides = node.get_input(2);
+
+    auto ksize_constant = get_constant_from_source(ksize);
+    TENSORFLOW_OP_VALIDATION(node, ksize_constant, "MaxPoolV2 is supported only with constant ksize.");
+    auto strides_constant = get_constant_from_source(strides);
+    TENSORFLOW_OP_VALIDATION(node, ksize_constant, "MaxPoolV2 is supported only with constant strides.");
 
-    int N = 2;
-    if (node.get_op_type() == "MaxPool3D") {
-        N = 3;
+    auto ksize_vector = ksize_constant->cast_vector<int64_t>();
+    auto strides_vector = strides_constant->cast_vector<int64_t>();
+
+    return translate_max_pool_util(node, 2, ksize_vector, strides_vector);
+}
+
+OutputVector translate_max_pool_op(const NodeContext& node) {
+    if (node.get_op_type() == "MaxPool") {
+        return translate_max_pool(node, 2);
+    } else if (node.get_op_type() == "MaxPoolV2") {
+        return translate_max_pool_v2(node);
+    } else if (node.get_op_type() == "MaxPool3D") {
+        return translate_max_pool(node, 3);
+    } else {
+        TENSORFLOW_OP_VALIDATION(node, false, "Only MaxPool2D, MaxPoolV2 and MaxPool3D are supported.");
```

Comment:
please add "node.get_op_type()" to this log

Reply:
Will be fixed in the next PR. It is not a blocker now to merge since CI is too-too slow.